pkg/apis/numaplane/v1alpha1/status_types.go

```diff
@@ -201,3 +201,16 @@ func (s *Status) markTypeStatus(t ConditionType, status metav1.ConditionStatus,
 		ObservedGeneration: generation,
 	})
 }
+
+// IsReady returns true when all the conditions are true
+func (s *Status) IsReady() bool {
```

Comment:
when we consider rollout object isHealthy, we only need to consider if the deployment/orchestration is successful. we definitely not want to include if the underlying pipeline is healthy or not. because numaflow already expose metrics for that. So I think we only need to consider `ChildResourceDeployed` condition.

Reply:
In that case checking only `phase` is enough, I believe we don't need to check for condition `ChildResourceDeployed` as phase status condition as getting set at same time [here](https://github.com/numaproj/numaplane/blob/main/pkg/apis/numaplane/v1alpha1/status_types.go#L138)